code/game/objects/items/devices/radio_jammer.dm

```diff
@@ -1,59 +1,87 @@
 //Global list for housing active radiojammers:
 var/list/active_radio_jammers = list()
 
-proc/within_jamming_range(var/atom/test) // tests if an object is near a radio jammer
-	if (active_radio_jammers && active_radio_jammers.len)
-		for (var/obj/item/device/radiojammer/Jammer in active_radio_jammers)
-			if (get_dist(test, Jammer) <= Jammer.radius)
-				return 1
-
-	return 0
+// tests if an object is near a radio jammer
+// if need_all_blocked is false, the jammer only needs to be on JAMMER_SYNTHETIC to work
+/proc/within_jamming_range(var/atom/test, var/need_all_blocked = TRUE)
+	if(length(active_radio_jammers))
+		var/turf/our_turf = get_turf(test)
+		for(var/obj/item/device/radiojammer/J in active_radio_jammers)
+			var/turf/jammer_turf = get_turf(J)
+			if(our_turf.z != jammer_turf.z)
+				continue
+			if(get_dist(our_turf, jammer_turf) <= J.radius)
+				if(need_all_blocked && J.active != JAMMER_ALL)
+					continue
+				return TRUE
+	return FALSE
 
 /obj/item/device/radiojammer
 	name = "radio jammer"
 	desc = "A small, inconspicious looking item with an 'ON/OFF' toggle."
+	desc_info = "Use in-hand to activate or deactivate, alt-click while adjacent or in-hand to toggle whether it blocks all wireless signals, or just stationbound wireless interfacing."
 	icon = 'icons/obj/device.dmi'
 	icon_state = "shield0"
 	w_class = ITEMSIZE_SMALL
-
-	var/active = 0
+	var/selected_active_level = JAMMER_ALL
+	var/active = JAMMER_OFF
 	var/radius = 7
 	var/icon_state_active = "shield1"
 	var/icon_state_inactive = "shield0"
 
+/obj/item/device/radiojammer/active
+	active = JAMMER_ALL
+
 /obj/item/device/radiojammer/New()
 	..()
-	update()
+	update_icon()
 
 /obj/item/device/radiojammer/Destroy()
-	if (active)
-		active_radio_jammers -= src
+	active_radio_jammers -= src
 	return ..()
 
+/obj/item/device/radiojammer/examine(mob/user, distance)
+	. = ..()
+	if(selected_active_level != JAMMER_BINARY && Adjacent(user))
+		var/output_message = "\The [src] will block <b>[selected_active_level == JAMMER_ALL ? "all" : "synthetic"]</b> wireless radio signals."
+		to_chat(user, SPAN_NOTICE(output_message))
+
+/obj/item/device/radiojammer/AltClick(mob/user)
+	if(!Adjacent(user) || selected_active_level == JAMMER_BINARY)
+		return
+	var/choice = alert(user, "Do you want \the [src] to block all wireless signals, or only remote synthetic (AI, borgs) signals.", "Signal Selection", "All", "Only Synthetic")
+	if(!choice)
+		return
+	if(choice == "All")
+		selected_active_level = JAMMER_ALL
+	else
+		selected_active_level = JAMMER_SYNTHETIC
+	if(active)
+		active = selected_active_level
+	to_chat(user, SPAN_NOTICE("You set \the [src] to block [lowertext(choice)] wireless signals."))
```

Comment:
It would imho be nicer to have a small UI for that, that can be used to control what should be jammed.